ecs-init/docker/dependencies.go

```diff
@@ -157,6 +241,9 @@ func isNetworkError(err error) bool {
 func isRetryablePingError(err error) bool {
 	godockerError, ok := err.(*godocker.Error)
 	if ok {
+		if godockerError.Status == http.StatusBadRequest {
```

Comment:
when provided with an unsupported docker api version, docker returns "Bad Request" (status code 400), so don't get into a retry loop if the API version is not supported (client errors in general probably shouldn't be retried at all, but we'll just stick with 400 for now to avoid changing behavior more than necessary).